### PR TITLE
[microTVM] Update nrfjprog on reference virtual machine

### PR DESCRIPTION
**Pull request by mehrdadh**
This PR makes following changes:

- It updates nrfjprog on reference machine to version 10.12.1 which fixes programming on nrf5340dk board.
- Adds test for nrf5340dk board to RVM integration tests.

### CODE REVIEW

**Comment by mehrdadh** [comment]
@areusch please take a look when you get a chance. Thanks!

File: apps/microtvm/reference-vm/zephyr/base-box/test-config.json [hunk 1]
```diff
@@ -1,4 +1,12 @@
-{"vid_hex": "0483",
- "pid_hex": "374b",
- "test_cmd": ["pytest", "tests/micro/qemu/test_zephyr.py", "--microtvm-platforms=stm32f746xx"]
+{
+    "stm32f746xx": {
+        "vid_hex": "0483",
+        "pid_hex": "374b",
+        "test_cmd": ["pytest", "tests/micro/qemu/test_zephyr.py", "--microtvm-platforms=stm32f746xx"]
+        },
+    "nrf5340dk": {
```
Comment (areusch) [inline apps/microtvm/reference-vm/zephyr/base-box/test-config.json:7]: @mdw-octoml could you just confirm you see these USB vid/pid?

Comment (mdw-octoml) [reply apps/microtvm/reference-vm/zephyr/base-box/test-config.json:7]: Confirmed!

**Review by areusch — APPROVED** [review]

**Comment by areusch** [comment]
thanks @mehrdadh !

**Pull request merged**